include/vast/Dialect/Core/CoreLazy.td

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022-present, Trail of Bits, Inc.
+#ifndef VAST_DIALECT_CORE_CORELAZY
+#define VAST_DIALECT_CORE_CORELAZY
+
+include "mlir/IR/OpBase.td"
+
+include "mlir/IR/FunctionInterfaces.td"
+
+include "mlir/Interfaces/CallInterfaces.td"
+include "mlir/Interfaces/CastInterfaces.td"
+include "vast/Interfaces/SymbolInterface.td"
+
+def ValueRegion : Region< HasOneBlock, "value region" >;
+
+class lazyEval< string mnemonic, list < Trait > traits = [] >
+    : Core_Op< mnemonic, !listconcat(traits, []) >, Results<(outs AnyType:$result)>
+{
+    let summary = "lazy evaluator";
+    let description = [{ VAST lazy evaluator }];
+}
+
+def Core_LazyOp : lazyEval< "lazy.op", [] > {
+    let summary = "Lazily evaluate a region.";
+    let description = [{
+        An operation that marks that the region it contains should be evalutated only when necessary.
```

Comment:
Reword to "The operation servers to encapsulate delayed evaluation in its region."